content/post/language_migrations.md

```diff
@@ -0,0 +1,512 @@
+
+**Python** (16.1 %) appears to be the most attractive language, followed closely by **Java** (15.3 %). However, it's especially interesting since only 11.3 % of all source code written on GitHub, among those 22 languages is **Python**.
+
+In Erik's ranking, **Go** was the big winner with 16.4 %. Since Erik based his approach on Google queries, it seems that the buzz around Go that makes people wonder explicitly in blogs if they should move to this language, takes a bit of time to become apparent in terms of projects effectively written in **Go** on GitHub.
+
+Furthermore **C** (9.2 %) is doing well like in Erik's grading with 14.3 %, but it is due to the amount of projects coded in **C** on GitHub.
+
+Although there are ten times more lines of code on GitHub in **PHP** than in **Ruby**, we have the same probability to switch for those two.
+
+**Go** (3.2 %) appears on 9th position which is largely honorable given the small proportion (0.9 %) of **Go** projects which are written on GitHub. The same proportion of projects are run in **Perl** for example, but this language doesn't really awake passion (2 % popularity).
+
+### What about sticking to a language ?
+
+If we keep the main diagonal of our transition matrix before applying the power iteration, we obtain slightly different results. It mainly decreases top languages popularity while raising smaller ones'. Indeed, it seems logical to believe that developers who invest their time in mastering a language on the fringe of the others tend to stick to them unlike popular ones.
+
+In the rest of the post, we will consider our first representation of the dominant eigenvector we gave.
+
+### Back to the transition matrix
+
+What if we sort our transition matrix so that the most popular languages appear at the bottom, and compare it to Erik's.
+
+<div class="grid2x">
+<div>
+<div>
+<img src="/post/language_migrations/sum_matrix_22lang_eig.svg">
+<p align="center" class="dt"><pre><code class="hljs python">source{d}'s sorted transition matrix</code></pre></p>
+</div>
+</div>
+<div>
+<div>
+<img src="/post/language_migrations/erik_green.png">
+<p align="center" class="dt"><pre><code class="hljs python">Erik's sorted transition matrix</code></pre></p>
+</div>
+</div>
+</div>
+
+* Developers coding in one of the 5 most popular languages (**Java**, **C**, **C++**, **PHP**, **Ruby**) are more likely to switch to **Python** with around that 22 % of chance in average.
+
+* Also according to Erik's matrix, in case we use **Ojective-C**, we would switch to **Swift** and reversely with the greatest probability (24 and 19 %).
+
+* Similarly, a **Visual Basic** developer has more chance (24 %) to move to **C#**, when Erik's reaches the nearly certitude with 92 %.
+
+* GitHub users of **Clojure**, **C#** and above all **Scala** would rather switch to **Java** with respectively 22, 29 and 40 % of chance which is the highest probability.
```

Comment:
~GitHub~ Users of **Clojure**, **C#** and **Scala** **above all** ... 40% ~of~ chance which <what? rephrase, did not get you>